utils/nctl/sh/scenarios/emergency_upgrade_test_balances.sh

```diff
@@ -0,0 +1,170 @@
+source "$NCTL"/sh/node/svc_"$NCTL_DAEMON_TYPE".sh
+
+# Exit if any of the commands fail.
+set -e
+
+#######################################
+# Runs an integration tests that performs an emergency restart on the network.
+# It also simulates social consensus on replacing the original validators (nodes 1-5)
+# with a completely new set (nodes 6-10).
+#
+# Arguments:
+#   `timeout=XXX` timeout (in seconds) when syncing. Default=300 seconds.
+#   `version=X_Y_Z` new protocol version to upgrade to. Default=2_0_0.
+#######################################
+function main() {
+    log "------------------------------------------------------------"
+    log "Emergency upgrade test begins"
+    log "------------------------------------------------------------"
+
+    do_await_genesis_era_to_complete
+
+    local SRC_KEY=$(get_account_key "user" 1)
+    local SRC_PURSE=$(get_main_purse_uref ${SRC_KEY})
+    local INITIAL_SRC_BALANCE=$(get_account_balance ${SRC_PURSE})
+
+    log "Src key = ${SRC_KEY}"
+    log "Src purse = ${SRC_PURSE}"
+    log "Initial src balance = ${INITIAL_SRC_BALANCE}"
+
+    # 1. Wait until they're all included in the chain.
+    do_await_one_era
+    # 2. Stop the network for the emergency upgrade.
+    do_stop_network
+    # 3. Prepare the nodes for the upgrade.
+    do_prepare_upgrade
+    # 4. Restart the network (start both the old and the new validators).
+    do_restart_network
+    # 5. Wait for the network to upgrade.
+    do_await_network_upgrade
+    # 6. Wait until they're all included in the chain.
+    do_await_one_era
+
+    local TGT_KEY="010101010101010101010101010101010101010101010101010101010101010101"
+    local TGT_PURSE=$(get_main_purse_uref ${TGT_KEY})
+    local FINAL_SRC_BALANCE=$(get_account_balance ${SRC_PURSE})
+    local FINAL_TGT_BALANCE=$(get_account_balance ${TGT_PURSE})
+
+    local SRC_DIFF=$((INITIAL_SRC_BALANCE - FINAL_SRC_BALANCE))
+
+    log "Transferred $TRANSFER_AMOUNT"
+    log "Source: final balance $FINAL_SRC_BALANCE, diff $SRC_DIFF"
+    log "Target: final balance $FINAL_TGT_BALANCE"
+
+    if [ $FINAL_TGT_BALANCE -ne $TRANSFER_AMOUNT ]; then
+        log "FAILED: target final balance differs from the transfer amount."
+        exit 1
+    fi
+
+    if [ $SRC_DIFF -lt $TRANSFER_AMOUNT ]; then
+        log "FAILED: source balance changed by less than the transfer amount."
+        exit 1
+    fi
+
+    log "------------------------------------------------------------"
+    log "Emergency upgrade test ends"
+    log "------------------------------------------------------------"
+}
+
+function log_step() {
+    local COMMENT=${1}
+    log "------------------------------------------------------------"
+    log "STEP $STEP: $COMMENT"
+    log "------------------------------------------------------------"
+    STEP=$((STEP + 1))
+}
+
+function do_await_genesis_era_to_complete() {
+    log_step "awaiting genesis era to complete"
+    while [ "$(get_chain_era)" != "1" ]; do
```

Comment:
This I actually copied from another upgrade test, and I think it might be intentional - if we go with your suggestion and start the test _before_ the genesis era starts, `get_chain_era` will probably return some kind of error, which is not `"0"`, so the loop will break. It will work correctly with the current code, though.

Reply:
Oh - I see.  I asked for the change as it allowed me to complete the test when my nctl network was already at era 2 when the test started.

I guess if other tests are using the same mechanism as this, it's fine.  Although now that I've searched for `"$(get_chain_era)"`, I see a bunch of different ways to achieve this.  Maybe @momipsl could chime in and we could unify the approach across all these places (not necessarily in this PR though)

Reply:
It would return `'N/A'`.